src/Microsoft.Data.SqlClient/tests/FunctionalTests/SqlConnectionStringBuilderTest.cs

```diff
@@ -402,6 +402,25 @@ public void ConnectionBuilderEncryptBackwardsCompatibility()
             Assert.True(builder.Encrypt);
         }
 
+        [Fact]
+        public void EncryptParserValidValuesPropertyIndexerForEncryptionOption()
+        {
+            SqlConnectionStringBuilder builder = new();
+            try
+            {
+                builder["Encrypt"] = SqlConnectionEncryptOption.Strict;
+                CheckEncryptType(builder, SqlConnectionEncryptOption.Strict);
+                builder["Encrypt"] = SqlConnectionEncryptOption.Optional;
+                CheckEncryptType(builder, SqlConnectionEncryptOption.Optional);
+                builder["Encrypt"] = SqlConnectionEncryptOption.Mandatory;
+                CheckEncryptType(builder, SqlConnectionEncryptOption.Mandatory);
+            }
+            catch (SqlException ex)
+            {
+                Assert.Fail(ex.Message);
+            }
```

Comment:
Unexpected exceptions don't need to be handled. Let the exception be thrown with the most information.
```suggestion
            builder["Encrypt"] = SqlConnectionEncryptOption.Strict;
            CheckEncryptType(builder, SqlConnectionEncryptOption.Strict);
            builder["Encrypt"] = SqlConnectionEncryptOption.Optional;
            CheckEncryptType(builder, SqlConnectionEncryptOption.Optional);
            builder["Encrypt"] = SqlConnectionEncryptOption.Mandatory;
            CheckEncryptType(builder, SqlConnectionEncryptOption.Mandatory);
```